pkg/storage/unified/apistore/prepare.go

```diff
@@ -39,30 +39,35 @@ func formatBytes(numBytes int) string {
 }
 
 // Called on create
-func (s *Storage) prepareObjectForStorage(ctx context.Context, newObject runtime.Object) ([]byte, error) {
+func (s *Storage) prepareObjectForStorage(ctx context.Context, newObject runtime.Object) ([]byte, string, error) {
 	info, ok := authtypes.AuthInfoFrom(ctx)
 	if !ok {
-		return nil, errors.New("missing auth info")
+		return nil, "", errors.New("missing auth info")
 	}
 
 	obj, err := utils.MetaAccessor(newObject)
 	if err != nil {
-		return nil, err
+		return nil, "", err
 	}
 	if obj.GetName() == "" {
-		return nil, storage.NewInvalidObjError("", "missing name")
+		return nil, "", storage.NewInvalidObjError("", "missing name")
 	}
 	if obj.GetResourceVersion() != "" {
-		return nil, storage.ErrResourceVersionSetOnCreate
+		return nil, "", storage.ErrResourceVersionSetOnCreate
 	}
 	if obj.GetUID() == "" {
 		obj.SetUID(types.UID(uuid.NewString()))
 	}
 	if obj.GetFolder() != "" && !s.opts.EnableFolderSupport {
-		return nil, apierrors.NewBadRequest(fmt.Sprintf("folders are not supported for: %s", s.gr.String()))
+		return nil, "", apierrors.NewBadRequest(fmt.Sprintf("folders are not supported for: %s", s.gr.String()))
+	}
+
+	grantPermisions := obj.GetAnnotation(utils.AnnoKeyGrantPermissions)
```

Comment:
This is to avoid someone externally passing this in, right?

Do we want to add explicit unit test maybe, considering this is a security measure against malicious requests?

And potentially throw an explicit error rather than cleaning it up?

Reply:
That's not it. This prevents storing the annotation in the final resource to avoid having the annotation showing up in future updates. 

Added a unit-test.